R/bar_plot_emission_profile_financial.R

```diff
@@ -0,0 +1,106 @@
+#' benchmark(s) to be plotted.
+#' @param mode The mode of financial data to plot.
+#' It can be one of "equal_weight", "worst_case" or "best_case".
+#' If nothing is chosen, "equal_weight" is the default case.
+#'
+#' @return A [ggplot] object.
+#' @export
+#'
+#' @examples
+#' benchmarks <- c("all", "unit", "isic_4digit")
+#' bar_plot_emission_profile_financial(financial, benchmarks, "equal_weight")
+bar_plot_emission_profile_financial <- function(data,
+                                                benchmarks = c(
+                                                  "all",
+                                                  "isic_4digit",
+                                                  "tilt_sector",
+                                                  "unit",
+                                                  "unit_isic_4digit",
+                                                  "unit_tilt_sector"
+                                                ),
+                                                mode = c(
+                                                  "equal_weight",
+                                                  "worst_case",
+                                                  "best_case"
+                                                )) {
+  benchmarks_arg <- arg_match(benchmarks, multiple = TRUE)
+  mode <- arg_match(mode)
+
+  crucial <- c(
+    "amount_total",
+    "bank_id",
+    "company_name",
+    "emission_profile",
+    "benchmark",
+    "equal_weight_finance",
+    "worst_case_finance",
+    "best_case_finance"
+  )
+  data |> check_crucial_names(names_matching(data, crucial))
+
+  risk_var <- names_matching(data, "emission_profile")
+
+  data <- data |>
+    mutate(risk_category_var = as_risk_category(data[[risk_var]]))
+
+  mode_col <- switch_mode(mode)
+
+  data <- data |>
+    calc_benchmark_emission_profile_financial(
+      risk_var,
+      benchmarks_arg,
+      mode_col
+    )
+
+  ggplot(data, aes(x = .data$percentage_total, y = .data$benchmark, fill = .data$risk_category_var)) +
+    geom_col(position = position_stack(reverse = TRUE), width = width_bar()) +
+    fill_score_colors() +
+    theme_tiltplot() +
+    xlim(0, 1)
+}
+
+#' Calculate emission profile proportions for specific benchmarks, with financial
+#' data
+#'
+#' @param data A data frame containing the emission profile data.
+#' @param risk_var The name of the variable containing risk categories.
+#' @param benchmarks A character vector specifying the benchmark(s) to consider.
+#' @param mode A character vector specifying the financial case to consider.
+#'
+#' @return A data frame with calculated proportions of emission profile categories.
+#'
+#' @examples
+#' calc_benchmark_emission_profile_financial(
+#'   financial,
+#'   "emission_profile",
+#'   c("all", "unit"),
+#'   "equal_weight"
+#' )
+#' @noRd
+calc_benchmark_emission_profile_financial <- function(data,
+                                                      risk_var,
```

Comment:
It seems that the suffix _var and the suffix _col both refer to a column in the the dataframe. If so maybe pick one or the other suffix and stick with it?